### PR TITLE
fix: add range validation to AddJournalTodoDtoValidator.TimezoneOffsetMinutes

### PR DESCRIPTION
**Pull request by nevridge**
## Summary
- Closes #185
- `AddJournalTodoDtoValidator` previously only validated `TaskItemId`, leaving `TimezoneOffsetMinutes` completely unchecked despite it feeding directly into the "today" calculation used by `JournalEntryRepository.AddTodoAsync`'s past-day-not-allowed rule. A garbage value (e.g. `999999`) was silently accepted and produced a nonsensical "today".
- Added `.InclusiveBetween(-720, 840)` (UTC-12:00 to UTC+14:00) applied via `.When(x => x.TimezoneOffsetMinutes.HasValue)`, so `null` (no offset supplied) remains valid.
- No API contract change — same endpoint/request/response shape; out-of-range values now correctly return `400 Bad Request` instead of corrupting the past-day check.
- `CreateTaskItemDto`/`UpdateTaskItemDto` share the same unvalidated field but are explicitly out of scope for this issue (separate tech debt).

## Test plan
- [x] Added `Validate_ShouldFail_WhenTimezoneOffsetMinutesIsOutOfRange` (`-721`, `841`, `999999`)
- [x] Added `Validate_ShouldPass_WhenTimezoneOffsetMinutesIsWithinRange` (`-720`, `840`, `0`)
- [x] Added `Validate_ShouldPass_WhenTimezoneOffsetMinutesIsNull`
- [x] `dotnet build` succeeds
- [x] `dotnet test` — 261 passed, 0 failed
- [x] `dotnet format --verify-no-changes` — no violations
- [x] implementation-validator spec compliance review — PASS, 0 critical/important/minor findings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs [hunk 1]
```diff
@@ -8,5 +8,10 @@ public class AddJournalTodoDtoValidator : AbstractValidator<AddJournalTodoDto>
     public AddJournalTodoDtoValidator()
     {
         RuleFor(x => x.TaskItemId).GreaterThan(0).WithMessage("TaskItemId must be greater than 0.");
+
+        RuleFor(x => x.TimezoneOffsetMinutes)
+            .InclusiveBetween(-720, 840)
+            .WithMessage("TimezoneOffsetMinutes must be between -720 and 840 (UTC-12:00 to UTC+14:00).")
+            .When(x => x.TimezoneOffsetMinutes.HasValue);
```
Comment (nevridge) [inline TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs]: `.When(x => x.TimezoneOffsetMinutes.HasValue)` is redundant here. Verified against a live repro on FluentValidation 12.1.1 (the version this repo pins): `InclusiveBetween` on a nullable property already returns valid for `null` with no `.When()` guard needed — removing this line produces identical results for null/in-range/out-of-range inputs. Simpler form: `.InclusiveBetween(-720, 840).WithMessage(...)` with no trailing `.When()`.

File: TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs [hunk 2]
```diff
@@ -8,5 +8,10 @@ public class AddJournalTodoDtoValidator : AbstractValidator<AddJournalTodoDto>
     public AddJournalTodoDtoValidator()
     {
         RuleFor(x => x.TaskItemId).GreaterThan(0).WithMessage("TaskItemId must be greater than 0.");
+
+        RuleFor(x => x.TimezoneOffsetMinutes)
+            .InclusiveBetween(-720, 840)
```
Comment (nevridge) [inline TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs:13]: Minor: `-720`/`840` is a raw magic-number literal with no shared constant. `CreateTaskItemDto`/`UpdateTaskItemDto` carry the same `TimezoneOffsetMinutes` field feeding identical `AddMinutes` arithmetic in `TaskItemsController.cs` but have no validator yet — when one is added there, this bound/message will likely get copy-pasted rather than shared, risking drift on a future correction. Consider a small shared constant or validator extension if/when that follow-up lands.

File: TaskFlow.Api.Tests/Validators/AddJournalTodoDtoValidatorTests.cs [hunk 3]
```diff
@@ -32,4 +32,37 @@ public async Task Validate_ShouldPass_WhenTaskItemIdIsPositive(int id)
 
         result.IsValid.Should().BeTrue();
     }
+
+    [Theory]
+    [InlineData(-721)]
+    [InlineData(841)]
+    [InlineData(999999)]
```
Comment (nevridge) [inline TaskFlow.Api.Tests/Validators/AddJournalTodoDtoValidatorTests.cs:39]: Minor: `InlineData(999999)` doesn't add coverage beyond the adjacent boundary case `841` — both exercise the same `value > max` rejection path. `-721` and `841` alone already fully pin the boundary behavior.

**Review by nevridge — COMMENTED** [review]
Automated review pass (8-angle finder + verification). One finding falls outside this diff (TaskItemsController.cs is not a changed file in this PR) so it's posted as a general comment below rather than inline; the rest are inline on the changed lines.

**Not part of this diff, flagged for follow-up:** `TaskFlow.Api/Controllers/V1/TaskItemsController.cs` line 90 (`Create`) and lines ~240-244 (`Update`) — `CreateTaskItemDto`/`UpdateTaskItemDto.TimezoneOffsetMinutes` feed the identical `AddMinutes` date arithmetic that issue #185 fixed for `AddJournalTodoDto`, but have no validator at all (confirmed: `TaskItemValidator` only validates `Title`/`Status`/`Priority` on the `TaskItem` model, not these DTOs). A garbage offset there can still throw `ArgumentOutOfRangeException` (unhandled 500) or corrupt the past-day check — same defect class, unfixed. The PR description calls this out as intentionally out-of-scope tech debt, which is a reasonable call for a narrow fix, but worth prioritizing as a correctness follow-up rather than "someday."

File: TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs (hunk 1 above)
Comment (nevridge) [reply TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs]: Confirmed — verified with a standalone repro against FluentValidation 12.1.1 (the pinned version): `InclusiveBetween` on a nullable `int?` returns valid for `null` with no `.When()` guard needed. Removed the redundant `.When(x => x.TimezoneOffsetMinutes.HasValue)` in eadb5b1. All 12 validator tests plus the full suite (261 tests) still pass.

File: TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs (hunk 2 above)
Comment (nevridge) [reply TaskFlow.Api/Validators/AddJournalTodoDtoValidator.cs:13]: Pushing back on this one. Agreed a shared constant would make sense once a second validator needs it, but that validator doesn't exist yet — CreateTaskItemDto/UpdateTaskItemDto currently have no TimezoneOffsetMinutes validation at all, and this PR's description explicitly scopes that out as separate tech debt (tracked separately from #185). Adding a shared constant/extension now for a consumer that doesn't exist would be speculative — nothing today shares this bound. Happy to factor it out in whichever PR adds the second validator, when there's an actual duplication to eliminate rather than a hypothetical one.

File: TaskFlow.Api.Tests/Validators/AddJournalTodoDtoValidatorTests.cs (hunk 3 above)
Comment (nevridge) [reply TaskFlow.Api.Tests/Validators/AddJournalTodoDtoValidatorTests.cs:39]: Pushing back on this one. Agreed -721/841 alone fully pin the boundary from a code-path/branch-coverage standpoint. But 999999 isn't there for path coverage — it's the literal repro value from issue #185's bug report ('A garbage value (e.g. 999999) is silently accepted'). Keeping it as its own InlineData case documents that this specific reported scenario is covered, independent of whatever the boundary values happen to be. I'd rather keep the direct traceability to the original bug report than trim it for a marginal reduction in test rows.
